osutil/mountinfo.go

```diff
@@ -46,6 +48,42 @@ type MountInfoEntry struct {
 	SuperOptions   map[string]string
 }
 
+func flattenMap(m map[string]string) string {
+	keys := make([]string, 0, len(m))
+	for key := range m {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	var buf bytes.Buffer
+	for i, key := range keys {
+		if i > 0 {
+			buf.WriteString(",")
+		}
+		if m[key] != "" {
+			fmt.Fprintf(&buf, "%s=%s", key, m[key])
+		} else {
+			buf.WriteString(key)
+		}
+	}
+	return buf.String()
+}
+
+func flattenList(l []string) string {
+	return strings.Join(l, ",")
+}
+
+func (mi *MountInfoEntry) String() string {
+	maybeSpace := " "
+	if len(mi.OptionalFields) == 0 {
+		maybeSpace = ""
+	}
+	return fmt.Sprintf("%d %d %d:%d %s %s %s %s%s- %s %s %s",
+		mi.MountID, mi.ParentID, mi.DevMajor, mi.DevMinor, escape(mi.Root),
+		escape(mi.MountDir), flattenMap(mi.MountOptions), flattenList(mi.OptionalFields),
+		maybeSpace, mi.FsType, escape(mi.MountSource),
+		flattenMap(mi.SuperOptions))
```

Comment:
You're escaping the strings in this fmt.Sprintf, but aren't in the flattenList and flattenMap functions. Shouldn't you?

Reply:
Ah, good catch. Done and tested now.